spec/datadog/di/instrumenter_spec.rb

```diff
@@ -0,0 +1,641 @@
+require "datadog/di/spec_helper"
+require 'datadog/di'
+require_relative 'hook_line'
+require_relative 'hook_method'
+
+RSpec.describe Datadog::DI::Instrumenter do
+  di_test
+
+  let(:observed_calls) { [] }
+
+  let(:settings) do
+    double('settings').tap do |settings|
```

Comment:
what do you think about using `instance_double` here instead? It has a benefit that the test will fail if stubbed methods are not available in the specified class.

Reply:
I think this is a good suggestion and after all of the pending DI code is merged I intend to go through the tests and replace `double` with `instance_double`. However, for settings,  I don't know what type to use for non-top-level settings since those use unnamed classes.